### ISSUE TITLE
[[lifetimebound]] basic members tracking

### ISSUE THREAD

**Issue by DenisYaroshevskiy** [llvm/llvm-project#81589]
Hi!

Would be amazing if the following is caught to:

https://godbolt.org/z/5ze9s5roj
```cpp
#include <utility>

struct Struct {
  std::pair<int, int> member;

  const auto* foo() const [[clang::lifetimebound]] { return &member; }
};

int main() {
  // auto& caught = *Struct().foo();
  auto& not_caught = Struct().foo()->second;  // this is a bug
  (void)not_caught;
}
```

**Comment by DenisYaroshevskiy** [comment]
Thank you very much! Looking forward fo rthis being available in my compiler

**Comment by usx95** [comment]
Created PR https://github.com/llvm/llvm-project/pull/100725

**Comment by usx95** [comment]
PR is merged now.
